FreeRTOS/FreeRTOS
# Fix issues in TCP QEMU Demo

Fix issues in TCP QEMU Demo

Description
-----------

1. **configKERNEL_INTERRUPT_PRIORITY** should be 255.
2. Ethernet ISR calls FreeRTOS APIs and therefore **configMAC_INTERRUPT_PRIORITY** should be equal to or greater than  configMAX_SYSCALL_INTERRUPT_PRIORITY.
3. The demo does not set the Ethernet interrupt priority and as a result, it remains at the highest priority. So the **Ethernet interrupt priority** is set to configMAC_INTERRUPT_PRIORITY .
4. **__NVIC_PRIO_BITS** set to 8 to match the QEMU implementation of PRIO bits.

Test Steps
-----------
TCP_Echo_QEMU_mps2 Demo is run with these modifications successfully.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
